tutel/experts/ffn.py

```diff
@@ -27,21 +31,51 @@ def update(self, ctx):
         local_experts = ctx.num_local_experts
         self.output_dim = self.output_dim or model_dim
 
-        fc1_weight = torch.empty(1, local_experts, hidden_size, model_dim)
-        fc2_weight = torch.empty(1, local_experts, hidden_size, self.output_dim)
-        fc1_bias = torch.empty(1, local_experts, hidden_size)
-        fc2_bias = torch.empty(1, local_experts, (self.output_dim + ctx.sharded_count - 1) // ctx.sharded_count)
-
-        for i in range(local_experts):
-            fc1 = torch.nn.Linear(model_dim, hidden_size)
-            fc2 = torch.nn.Linear(hidden_size, self.output_dim)
-            fc1_weight[0, i, :, :], fc1_bias[0, i, :] = fc1.weight, fc1.bias
-            fc2_weight[0, i, :, :], fc2_bias[0, i, :] = fc2.weight.t(), fc2.bias[:fc2_bias.size(-1)]
-
-        self.register_parameter(name='batched_fc1_w', param=torch.nn.Parameter(fc1_weight.squeeze(0)))
-        self.register_parameter(name='batched_fc2_w', param=torch.nn.Parameter(fc2_weight.squeeze(0)))
-        self.register_parameter(name='batched_fc1_bias', param=torch.nn.Parameter(fc1_bias.squeeze(0)))
-        self.register_parameter(name='batched_fc2_bias', param=torch.nn.Parameter(fc2_bias.squeeze(0)))
+        fc1_weight = torch.empty(local_experts, hidden_size, model_dim, dtype=dtype, device=device)
```

Comment:
The purpose of above code is to guarantee a deterministic weight initialization value just by providing manual_seed, which is shared by CPU and CUDA. What is the reason you have to change them?

Reply:
Is it possible not to change the original initialization?